common/formatting/layout_optimizer.h

```diff
@@ -0,0 +1,44 @@
+// Copyright 2017-2021 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Implementation details of layout_optimizer.cc exported for tests.
+
+#ifndef VERIBLE_VERILOG_FORMATTING_LAYOUT_OPTIMIZER_H_
+#define VERIBLE_VERILOG_FORMATTING_LAYOUT_OPTIMIZER_H_
+
+#include <algorithm>
+#include <iterator>
+#include <ostream>
+#include <type_traits>
+#include <vector>
+
+#include "absl/container/fixed_array.h"
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_join.h"
+#include "common/formatting/basic_format_style.h"
+#include "common/formatting/token_partition_tree.h"
+#include "common/formatting/unwrapped_line.h"
+#include "common/util/vector_tree.h"
+
+namespace verible {
+
+// Handles formatting of TokenPartitionTree nodes with
+// kOptimalFunctionCallLayout partition policy.
+void OptimizeTokenPartitionTree(TokenPartitionTree* node,
```

Comment:
This needs a bit more description of what happens with `node` (it will be formatted afterwards), what happens with `ftokens`, what is the expected outcome.

And maybe I'd fully qualify `PartitionPolicyEnum::kOptimalFunctionCallLayout`.

A function header description should always give the reader enough information to use the function just from the description.

The constant `style` parameter should be the [the first](https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs) parameter.